cmd/endpoint.go

```diff
@@ -142,6 +146,27 @@ func NewEndpoint(arg string) (ep Endpoint, e error) {
 	}, nil
 }
 
+// checkIPv4 - helper for validating if the provided arg is an ip address.
+func checkIPv4(ipAddress string) bool {
+	host, _, err := net.SplitHostPort(ipAddress)
+	if err != nil {
+		// If the error is anything other than missing port in address return false.
+		// If it is missing port in address we want to go ahead and parse the host string without address
+		// to check if it is an IP. Doing it this way allows us to not use regex to do a crude pattern matching
+		// for an IP address with port string.
+		// TODO: See if checkIPv4 can be reused where SplitHostPort is often called to check if it's an IP.
+		if !strings.Contains(err.Error(), "missing port in address") {
```

Comment:
Similar changes needs to be achieved for other places we are using this pattern..  But in this case alone we can simplify.